server/graphql/v1/TransactionInterface.js

```diff
@@ -179,10 +179,16 @@ const TransactionFields = () => {
     },
     createdByUser: {
       type: UserType,
-      resolve(transaction) {
-        // If it's a sequelize model transaction, it means it has the method getCreatedByUser
-        // otherwise we return null
+      async resolve(transaction, args, req) {
+        // We don't return the user if the transaction has been created by someone who wanted to remain anonymous
+        // This is very suboptimal. We should probably record the CreatedByCollectiveId (or better CreatedByProfileId) instead of the User.
         if (transaction && transaction.getCreatedByUser) {
+          const fromCollective = await transaction.getFromCollective();
+          if (fromCollective.isAnonymous && (!req.remoteUser || !req.remoteUser.isAdmin(transaction.CollectiveId)))
+            return {};
+          const collective = await transaction.getCollective();
+          if (collective.isAnonymous && (!req.remoteUser || !req.remoteUser.isAdmin(transaction.FromCollectiveId)))
+            return {};
```

Comment:
Have you considered dropping `createdByUser` altogether? I'm pretty sure we don't need it / can avoid it.

Reply:
Maybe, it just feels like out of scope for this PR. I wanted to keep everything else "as is".

Reply:
If this can avoid that kind of suboptimal code, this is worth it.